### PR TITLE
chore(ci): update windows runner to 2022

### PR DESCRIPTION
**Pull request by pront**
### Issue 
closes: https://github.com/vectordotdev/vector/issues/21468

### Test plan

* Windows unit tests workflow [passed](https://github.com/vectordotdev/vector/actions/runs/11297157969) with the new [runner](https://github.com/organizations/vectordotdev/settings/actions/github-hosted-runners/50).
* Testing the publish workflow:
  * Used this commit https://github.com/vectordotdev/vector/pull/21486/commits/6a5433143a4452d13b60184f0f8d543cd4218827
  * ...and triggered this https://github.com/vectordotdev/vector/actions/runs/11297760750

### CODE REVIEW

**Review by pront — COMMENTED** [review]
/ci-run-unit-windows

**Comment by pront** [comment]
Windows unit tests workflow [passed](https://github.com/vectordotdev/vector/actions/runs/11297157969) with the new [runner](https://github.com/organizations/vectordotdev/settings/actions/github-hosted-runners/50).

File: changelog.d/21486_windows_2022_build.feat.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+The Vector CI now uses
```
Comment (jszwedko) [inline changelog.d/21486_windows_2022_build.feat.md:1]: I think this sort of CI change is likely to just be noise to end-users (i.e. they won't care). Vector developers may care, but the changelog audience is end-users. I'd probably mark this as `no-changelog`.

**Review by jszwedko — APPROVED** [review]
🚀

**Pull request merged**